test/WebJobs.Script.Tests/FunctionMetadataManagerTests.cs

```diff
@@ -153,12 +159,18 @@ public void FunctionMetadataManager_GetsMetadata_FromFunctionProviders()
             mockFunctionProvider.Setup(m => m.FunctionErrors).Returns(mockFunctionErrors.ToImmutableDictionary());
 
             FunctionMetadataManager testFunctionMetadataManager = TestFunctionMetadataManager.GetFunctionMetadataManager(new OptionsWrapper<ScriptJobHostOptions>(_scriptJobHostOptions),
-                mockFunctionMetadataProvider.Object, new List<IFunctionProvider>() { mockFunctionProvider.Object }, new OptionsWrapper<HttpWorkerOptions>(_defaultHttpWorkerOptions), MockNullLoggerFactory.CreateLoggerFactory(), new TestOptionsMonitor<LanguageWorkerOptions>(TestHelpers.GetTestLanguageWorkerOptions()));
+                mockFunctionMetadataProvider.Object, new List<IFunctionProvider>() { mockFunctionProvider.Object }, new OptionsWrapper<HttpWorkerOptions>(_defaultHttpWorkerOptions), loggerFactory, new TestOptionsMonitor<LanguageWorkerOptions>(TestHelpers.GetTestLanguageWorkerOptions()));
             testFunctionMetadataManager.LoadFunctionMetadata();
 
             Assert.Equal(0, testFunctionMetadataManager.Errors.Count);
             Assert.Equal(1, testFunctionMetadataManager.GetFunctionMetadata(true).Length);
+            var traces = testLoggerProvider.GetAllLogMessages();
             Assert.Equal("anotherFunction", testFunctionMetadataManager.GetFunctionMetadata(true).FirstOrDefault()?.Name);
+
+            // Assert logging traces print out as expected
+            Assert.Equal(7, traces.Count);
+            Assert.Equal(2, traces.Count(t => t.FormattedMessage.Contains("Reading functions metadata from FunctionMetadataManager")));
+            Assert.Equal(2, traces.Count(t => t.FormattedMessage.Contains("1 functions found in FunctionMetadataManager from custom provider")));
```

Comment:
Is the "0 functions loaded" also logged? Can you include asserts for this as well in these tests? I'm still confused why the updated logs you show in the PR description shows a function was found, but 0 loaded.

Reply:
So the custom provider finds that one function is there which is why 1 function is found [here](https://github.com/Azure/azure-functions-host/blob/dev/src/WebJobs.Script/Host/FunctionMetadataManager.cs#L231). However, since the metadata array is empty it doesn't add it to the `functionMetadataList` which is what the loaded statement prints out at the end

Reply:
So in the scenario listed in the PR description, it finds a function through the custom provider [here.](https://github.com/Azure/azure-functions-host/blob/dev/src/WebJobs.Script/Host/FunctionMetadataManager.cs#L231) For the specific function I was testing, there was no metadata associated with it,  I suppose which is why it says that no functions are loaded and it doesn't add it to the `functionMetadataList`.

For the mock test, it actually does load the function metadata since we explicitly add it so it actually does load it. I will add that in the test.